### PR TITLE
[sailfish-webview] Multiple alert fix. Fixes JB#51913

### PR DESCRIPTION
**Pull request by mkenttala**

### CODE REVIEW

File: import/popups/PopupOpener.qml [hunk 1]
```diff
@@ -78,19 +80,13 @@ Timer {
         var winId = data.winId
         switch (topic) {
         case "embed:alert": {
```
Comment (rainemak) [inline import/popups/PopupOpener.qml:82]: Can this get called more than 2 times? Should there be a queue of data objects received from the embedlite?

File: import/popups/PopupOpener.qml [hunk 2]
```diff
@@ -189,6 +185,31 @@ Timer {
...
+                    "winId": data.winId,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+        })
+    }
+
+    // Handle pagestack busy change
+    function busyChanged() {
+        if (!pageStack.busy) {
```
Comment (rainemak) [inline import/popups/PopupOpener.qml:207]:
```suggestion
        if (!pageStack.busy && root.alertData) {
```

File: import/popups/PopupOpener.qml [hunk 3]
```diff
@@ -42,6 +42,8 @@ Timer {
         anchor: Notice.Center
     }
 
+    property var alertData
```
Comment (rainemak) [inline import/popups/PopupOpener.qml:45]: Worth creating new tasks to fix confirm and prompt as well as those should be similarly broken. Should checked at least.

File: import/popups/PopupOpener.qml (hunk 1 above)
Comment (mkenttala) [reply import/popups/PopupOpener.qml:82]: This won't be called simultaneously so no queue needed. Fix is only needed to wait for the previous dialog/page to fully close.

File: import/popups/PopupOpener.qml (hunk 3 above)
Comment (mkenttala) [reply import/popups/PopupOpener.qml:45]: Will check

File: import/popups/PopupOpener.qml [hunk 4]
```diff
@@ -189,6 +154,127 @@ Timer {
...
+                    "accepted": false,
+                    "checkvalue": dialog.checkboxValue
+                })
+            })
+        })
+    }
+
+    // Open login dialog
+    function login(data) {
+        var obj = pageStack.animatorPush(Qt.resolvedUrl("PasswordManagerDialog.qml"), {
```
Comment (rainemak) [inline import/popups/PopupOpener.qml:228]:
```suggestion
        pageStack.animatorPush(Qt.resolvedUrl("PasswordManagerDialog.qml"), {
```

Comment (rainemak) [reply import/popups/PopupOpener.qml:228]: Please add also testcase for https://github.com/sailfishos/sailfish-browser/blob/next/tests/manual/testwebprompts.html. Simple testcase function that requests few alerts, prompts, and confirms.

**Review by rainemak — APPROVED** [review]
Reviewing and testing. Please add manual testcase that runs alerts, confirms, and prompts in row (sailfish-browser).

**Comment by rainemak** [comment]
@atatarov @okodron review as well

**Pull request merged**